src/editor/editor.js

```diff
@@ -189,6 +193,42 @@ export default class Editor {
 
 mix( Editor, EmitterMixin );
 
+// @param {module:utils/config~Config} config
+// @param {Object} builtInConfig
+// @param {Object.<String,*>} builtInConfig.config
+// @param {Array.<module:core/plugin~Plugin>} availablePlugins
+function extendEditorConfig( config, builtInConfig, availablePlugins ) {
```

Comment:
This looks bad. It shouldn't require so much code in this class to set the defaults. It must be `Config`'s ability to easily define the defaults.

I also need to think what should extend what here. I'll let you know once I figure this out :D.